src/functions_framework/__init__.py

```diff
@@ -67,6 +72,24 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def typed(*args):
```

Comment:
Hmm I've been pondering for a while whether there's a way for us to provide typing here where like someone could hover over the definition of @typed and see that it's supposed to just take one, optional input type parameter instead of this kind of ambiguous *args situation.

I'm not sure it's possible though unless we enforce a named parameter like `@typed(input_type=MyType)`...

Can you think of anything?